libs/ui-lib/lib/cim/components/Hypershift/modals/ManageHostsModal.tsx

```diff
@@ -27,13 +27,63 @@ type ManageHostsModalProps = {
     nodePool: NodePoolK8sResource,
     nodePoolPatches: {
       op: string;
-      value: unknown;
+      value?: unknown;
       path: string;
     }[],
   ) => Promise<unknown>;
   agentMachines: AgentMachineK8sResource[];
 };
 
+const getPatches = (values: NodePoolFormValues, nodePool: NodePoolK8sResource) => {
+  const patches = [];
+
+  if (values.useAutoscaling && !nodePool.spec.autoScaling) {
+    patches.push(
+      { op: 'remove', path: '/spec/replicas' },
+      {
+        op: 'add',
+        value: {
+          min: values.autoscaling?.minReplicas,
+          max: values.autoscaling?.maxReplicas,
+        },
+        path: '/spec/autoScaling',
+      },
+    );
+  } else if (!values.useAutoscaling && !nodePool.spec.replicas) {
+    patches.push(
+      { op: 'remove', path: '/spec/autoScaling' },
+      {
+        op: 'add',
+        value: values.count,
+        path: '/spec/replicas',
+      },
+    );
+  }
+
+  if (values.count !== nodePool.spec.replicas) {
```

Comment:
if line 40 or 52 is executed, we will end up with unnecessary patches here. The whole `getPatches` func should be reworked to be more efficient

Reply:
I hope it's better now :slightly_smiling_face: